### PR TITLE
Adds `is_io` function

### LINKED ISSUE #309: Add `is_io` function
It should return `True` for `IO`, `IOResult`, and `RequiresContextIOResult`

### PR DESCRIPTION
**Pull request by thepabloaguilar**
I've implemented the function, wrote tests, wrote examples in documentation.
Unfortunately, I've imported `RequiresContextIOResult` class within `is_io` function to avoid circular import.

Closes #309

### CODE REVIEW

File: returns/functions.py [hunk 1]
```diff
@@ -167,3 +170,20 @@ def not_(function: Callable[..., bool]) -> Callable[..., bool]:
     def wrapped_function(*args, **kwargs) -> bool:  # noqa: WPS430
         return not function(*args, **kwargs)
     return wrapped_function
+
+
+def is_io(container: BaseContainer) -> bool:
```
Comment (sobolevn) [inline returns/functions.py:175]: Is it possible to move it to `pipeline.py`? I am asking because I want to get rid of this nested import.

Comment (sobolevn) [reply returns/functions.py:175]: Nope, we should add a new module for it.
We would add `is_context, `is_result`, and `is_future`.

Let's call it `instances.py` maybe?

Comment (thepabloaguilar) [reply returns/functions.py:175]: What about `containers.py`? Because all of those functions will do something with containers!

Comment (thepabloaguilar) [reply returns/functions.py:175]: @sobolevn ?

Comment (sobolevn) [reply returns/functions.py:175]: Containers is too general. We check if something is an `instance` of `io`, `future`, etc. So, `instances.py` looks better for me.

Comment (thepabloaguilar) [reply returns/functions.py]: `instances` is too general, as well, this topic is much abstract kkkkk. But it's, I'll use `instances.py`!

**Comment by sobolevn** [comment]
Can you please open new issue for the similar functions but for `future`, `context`, `result`?

**Comment by thepabloaguilar** [comment]
> Can you please open new issue for the similar functions but for `future`, `context`, `result`?

Sure!!

**Pull request merged**